yt/utilities/lib/pixelization_routines.pyx

```diff
@@ -1202,75 +1211,98 @@ def pixelize_sph_kernel_projection(
                 with gil:
                     PyErr_CheckSignals()
 
-            xiter[1] = yiter[1] = 999
+            xiter[1] = yiter[1] = ziter[1] = 999
```

Comment:
Very confused as to why this is initialised as 999?

Reply:
It's just an arbitrary flag value. If a particle doesn't need to be 'duplicated' in a given dimension to capture its (full) effects in the projection, the second iteration of that duplication loop is skipped. The lines setting `xiter[1]`, `yiter[1]`, and `ziter[1]` to different values if we need the duplication are set in the lines directly following the initialization. Those nested loops for the different dimensions are in [these lines](https://github.com/yt-project/yt/blob/0580c692ad36035a97fa3561cd34f891ad02b7c3/yt/utilities/lib/pixelization_routines.pyx#L1251-L1268).